stactools_threedep/stactools/threedep/metadata.py

```diff
@@ -0,0 +1,218 @@
+    def start_datetime(self) -> Union[datetime.datetime, None]:
+        """Returns the start datetime for this record.
+
+        This can be a while ago, since the national elevation dataset was
+        originally derived from direct survey data.
+        """
+        return _format_date(self.begdate)
+
+    @property
+    def end_datetime(self) -> Union[datetime.datetime, None]:
+        """Returns the end datetime for this record."""
+        return _format_date(self.enddate, end_of_year=True)
+
+    @property
+    def gsd(self) -> float:
+        """Returns the nominal ground sample distance from these metadata."""
+        if self.product == "1":
+            return 30
+        elif self.product == "13":
+            return 10
+        else:
+            raise NotImplementedError
+
+    def data_asset(self, base: str = DEFAULT_BASE) -> Asset:
+        """Returns the data asset (aka the tiff file)."""
+        return Asset(href=self._asset_href_with_extension(base, "tif"),
+                     title=self.title,
+                     description=self.description,
+                     media_type=MediaType.COG,
+                     roles=["data"])
+
+    def metadata_asset(self, base: str = DEFAULT_BASE) -> Asset:
+        """Returns the data asset (aka the tiff file)."""
+        return Asset(href=self._asset_href_with_extension(base, "xml"),
+                     media_type=MediaType.XML,
+                     roles=["metadata"])
+
+    def thumbnail_asset(self, base: str = DEFAULT_BASE) -> Asset:
+        """Returns the numbnail asset."""
+        return Asset(href=self._asset_href_with_extension(base, "jpg"),
+                     media_type=MediaType.JPEG,
+                     roles=["thumbnail"])
+
+    def derived_from_link(self, base: str = DEFAULT_BASE) -> Link:
+        """Returns the derived from link for this file."""
+        return Link("derived_from",
+                    self._asset_href_with_extension(base, "xml"))
+
+    @property
+    def projection_extension_dict(self) -> dict:
+        """Returns a dictionary of values to be applied to the projection extension."""
+        shape = [int(self.rowcount), int(self.colcount)]
+        transform = [
+            float(self.longres),
+            0.0,
+            float(self.minx),
+            0.0,
+            -float(self.latres),
+            float(self.maxy),
+            0.0,
+            0.0,
+            1.0,
+        ]
+        bbox = [
+            float(self.minx),
+            float(self.miny),
+            float(self.maxx),
+            float(self.maxy)
+        ]
+        centroid = {
+            "lon": (bbox[0] + bbox[2]) / 2.0,
+            "lat": (bbox[1] + bbox[3]) / 2.0,
+        }
+        geometry = mapping(box(*bbox))
+        return {
+            "epsg": THREEDEP_EPSG,
+            "shape": shape,
+            "transform": transform,
+            "bbox": bbox,
+            "centroid": centroid,
+            "geometry": geometry,
+        }
```

Comment:
The `projection` extension does have a few fields that I think should always be included in STAC metadata as it allows apps like stac-vrt, stackstac and OpenDataCube to do their magic without first hitting data files. 
These fields include `epsg`, `shape`, and `transform`.

However, I don't see much use for bbox in native coordinates (this can be calculated from transform) at all.  `centroid` and `geometry` may be useful for some small number of use cases, but I think they are generally unnecessary.

thoughts @lossyrob ?

Reply:
If I recall correctly I included `bbox`, `centroid`, and `geometry` because [`Extension.apply`](https://github.com/stac-utils/stactools/pull/81/files#diff-880b7ec34b4338db8c7e89e38252fcbe043576a8382c13db6a619ae47969792cR47) was complaining without them, and they weren't too expensive to provide. Need to check that, and if that's true maybe it make sense to relax `apply`.

Reply:
There should only be a problem if they are required fields. If that's the case, should open up issue in PySTAC